src/main/java/seedu/duke/command/cap/CapCommand.java

```diff
@@ -92,24 +95,38 @@ private double gradeConvert(String grade) {
      * cap of the user
      * @throws InvalidCapCalculatedException
      * When the attained cap is unimaginable
+     * @throws ModuleNotFoundException
+     * When the module called isn't found
      */
     private double calculateCap() throws InvalidCapCalculatedException, ModuleNotFoundException {
         String[] moduleList = ModuleManager.getModCodeList();
-        double mcGrade;
+        double mcGrade = 0;
         double sumMcGrade = 0;
         double sumMc = 0;
         double gradeValue;
+        int numberOfCsCuModules = 0;
+        int numberOfModules = 0;
         for (String i : moduleList) {
+            numberOfModules++;
             gradeValue = gradeConvert(ModuleManager.getModule(i).getModuleGrade());
-            if (gradeValue == 0) {
+            if (gradeValue == -1.0) {
                 throw new InvalidCapCalculatedException();
+            } else if (gradeValue >= 0) {
+                sumMc += ModuleManager.getModule(i).getModuleCredit();
+                mcGrade = ModuleManager.getModule(i).getModuleCredit()
+                        * gradeValue;
+                sumMcGrade += mcGrade;
+            } else {
+                numberOfCsCuModules++;
             }
-            sumMc += ModuleManager.getModule(i).getModuleCredit();
-            mcGrade = ModuleManager.getModule(i).getModuleCredit()
-                    * gradeValue;
-            sumMcGrade += mcGrade;
         }
+        if (numberOfCsCuModules == numberOfModules) {
+            sumMcGrade = 20;
```

Comment:
What's this for?

Reply:
checking if that semester they SU all the mods